react-app/src/api/cosmosAPI.ts

```diff
@@ -102,6 +113,61 @@ export const useCosmos = (): ICosmosAPI => {
     [chainInfo.currency.coinMinimalDenom, signTx, getBalance, wallet]
   );
 
+  const signDelegateTokenTx = useCallback(
```

Comment:
can we split these to own files, otherwise this file will grow very fast?

Reply:
shall we handle signing and broadcasting tx in `cosmosAPI.ts` and handle message making in a model file